app/views/admin/products_v3/_variant_row.html.haml

```diff
@@ -21,7 +21,7 @@
           value: unit_value_with_description(variant), 'aria-label': t('admin.products_page.columns.unit_value'), required: true
     .field
       = f.label :display_as, t('admin.products_page.columns.display_as')
-      = f.text_field :display_as, placeholder: VariantUnits::OptionValueNamer.new(variant).name
+      = f.text_field :display_as, 'aria-label': t('admin.products_page.columns.display_as'), placeholder: VariantUnits::OptionValueNamer.new(variant).name
```

Comment:
Above, `f.label :display_as` should produce a label with ID pointing to this text field. Therefore, the additional `aria-label` shouldn't be necessary. I think Capybara should be able to `find_field "Display unit as", placeholder: "1kg"` without any help.

Hmm, but the `find` in the spec isn't doing anything, it should be an expectation..

Reply:
Sorry about the aria-label. That's my bad.

> but the find in the spec isn't doing anything, it should be an expectation..

Ideally yes. I used the `find` method because it raises the exception `ElementNotFound` if it's unable to find the element otherwise It goes on.
But yes, expectation is better as it's more clear what the spec is doing. Thanks for rasing it. 👍